/*
 * SPDX-License-Identifier: MIT
 * SPDX-FileCopyrightText: Copyright (c) 2024 OpenStickCommunity (gp2040-ce.info)
 */

#ifndef PICO_BOARD_CONFIG_H_
#define PICO_BOARD_CONFIG_H_

#include "enums.pb.h"
#include "class/hid/hid.h"

#define BOARD_CONFIG_LABEL "SplitFoxPro"

// Main pin mapping Configuration
//                                                  // GP2040 | Xinput | Switch  | PS3/4/5  | Dinput | Arcade |
#define GPIO_PIN_28 GpioAction::BUTTON_PRESS_UP     // UP     | UP     | UP      | UP       | UP     | UP     |
#define GPIO_PIN_27 GpioAction::BUTTON_PRESS_DOWN   // DOWN   | DOWN   | DOWN    | DOWN     | DOWN   | DOWN   |
#define GPIO_PIN_29 GpioAction::BUTTON_PRESS_RIGHT  // RIGHT  | RIGHT  | RIGHT   | RIGHT    | RIGHT  | RIGHT  |
#define GPIO_PIN_26 GpioAction::BUTTON_PRESS_LEFT   // LEFT   | LEFT   | LEFT    | LEFT     | LEFT   | LEFT   |
#define GPIO_PIN_09 GpioAction::BUTTON_PRESS_B1     // B1     | A      | B       | Cross    | 2      | K1     |
#define GPIO_PIN_14 GpioAction::BUTTON_PRESS_B2     // B2     | B      | A       | Circle   | 3      | K2     |
#define GPIO_PIN_15 GpioAction::BUTTON_PRESS_R2     // R2     | RT     | ZR      | R2       | 8      | K3     |
#define GPIO_PIN_18 GpioAction::BUTTON_PRESS_L2     // L2     | LT     | ZL      | L2       | 7      | K4     |
#define GPIO_PIN_10 GpioAction::BUTTON_PRESS_B3     // B3     | X      | Y       | Square   | 1      | P1     |
#define GPIO_PIN_11 GpioAction::BUTTON_PRESS_B4     // B4     | Y      | X       | Triangle | 4      | P2     |
#define GPIO_PIN_12 GpioAction::BUTTON_PRESS_R1     // R1     | RB     | R       | R1       | 6      | P3     |
#define GPIO_PIN_13 GpioAction::BUTTON_PRESS_L1     // L1     | LB     | L       | L1       | 5      | P4     |
#define GPIO_PIN_24 GpioAction::BUTTON_PRESS_S1     // S1     | Back   | Minus   | Select   | 9      | Coin   |
#define GPIO_PIN_25 GpioAction::BUTTON_PRESS_S2     // S2     | Start  | Plus    | Start    | 10     | Start  |
#define GPIO_PIN_04 GpioAction::BUTTON_PRESS_L3     // L3     | LS     | LS      | L3       | 11     | LS     |
#define GPIO_PIN_22 GpioAction::BUTTON_PRESS_R3     // R3     | RS     | RS      | R3       | 12     | RS     |
#define GPIO_PIN_23 GpioAction::BUTTON_PRESS_A1     // A1     | Guide  | Home    | PS       | 13     | ~      |
#define GPIO_PIN_20 GpioAction::BUTTON_PRESS_A2     // A2     | ~      | Capture | Touchpad | 14     | ~      |

#define GPIO_PIN_19 GpioAction::BUTTON_PRESS_UP     // UP     | UP     | UP      | UP       | UP     | UP     |
#define GPIO_PIN_16 GpioAction::BUTTON_PRESS_UP     // UP     | UP     | UP      | UP       | UP     | UP     |
#define GPIO_PIN_17 GpioAction::BUTTON_PRESS_R3     // R3     | RS     | RS      | R3       | 12     | RS     |

#define GPIO_PIN_06 GpioAction::BUTTON_PRESS_A1     // A1     | Guide  | Home    | PS       | 13     | ~      |
#define GPIO_PIN_07 GpioAction::BUTTON_PRESS_S1     // S1     | Back   | Minus   | Select   | 9      | Coin   |
#define GPIO_PIN_08 GpioAction::BUTTON_PRESS_S2     // S2     | Start  | Plus    | Start    | 10     | Start  |

// Setting GPIO pins to assigned by add-on
//
#define GPIO_PIN_00 GpioAction::ASSIGNED_TO_ADDON //i2c sda
#define GPIO_PIN_01 GpioAction::ASSIGNED_TO_ADDON //i2c scl
#define GPIO_PIN_21 GpioAction::ASSIGNED_TO_ADDON //rgb
#define GPIO_PIN_05 GpioAction::SUSTAIN_FOCUS_MODE //focus
#define GPIO_PIN_02 GpioAction::ASSIGNED_TO_ADDON //USB_PERIPHERAL_PIN_DMINUS
#define GPIO_PIN_03 GpioAction::ASSIGNED_TO_ADDON //USB_PERIPHERAL_PIN_DPLUS

// Keyboard Mapping Configuration
//                                            // GP2040 | Xinput | Switch  | PS3/4/5  | Dinput | Arcade |
#define KEY_DPAD_UP     HID_KEY_ARROW_UP      // UP     | UP     | UP      | UP       | UP     | UP     |
#define KEY_DPAD_DOWN   HID_KEY_ARROW_DOWN    // DOWN   | DOWN   | DOWN    | DOWN     | DOWN   | DOWN   |
#define KEY_DPAD_RIGHT  HID_KEY_ARROW_RIGHT   // RIGHT  | RIGHT  | RIGHT   | RIGHT    | RIGHT  | RIGHT  |
#define KEY_DPAD_LEFT   HID_KEY_ARROW_LEFT    // LEFT   | LEFT   | LEFT    | LEFT     | LEFT   | LEFT   |
#define KEY_BUTTON_B1   HID_KEY_SHIFT_LEFT    // B1     | A      | B       | Cross    | 2      | K1     |
#define KEY_BUTTON_B2   HID_KEY_Z             // B2     | B      | A       | Circle   | 3      | K2     |
#define KEY_BUTTON_R2   HID_KEY_X             // R2     | RT     | ZR      | R2       | 8      | K3     |
#define KEY_BUTTON_L2   HID_KEY_V             // L2     | LT     | ZL      | L2       | 7      | K4     |
#define KEY_BUTTON_B3   HID_KEY_CONTROL_LEFT  // B3     | X      | Y       | Square   | 1      | P1     |
#define KEY_BUTTON_B4   HID_KEY_ALT_LEFT      // B4     | Y      | X       | Triangle | 4      | P2     |
#define KEY_BUTTON_R1   HID_KEY_SPACE         // R1     | RB     | R       | R1       | 6      | P3     |
#define KEY_BUTTON_L1   HID_KEY_C             // L1     | LB     | L       | L1       | 5      | P4     |
#define KEY_BUTTON_S1   HID_KEY_5             // S1     | Back   | Minus   | Select   | 9      | Coin   |
#define KEY_BUTTON_S2   HID_KEY_1             // S2     | Start  | Plus    | Start    | 10     | Start  |
#define KEY_BUTTON_L3   HID_KEY_EQUAL         // L3     | LS     | LS      | L3       | 11     | LS     |
#define KEY_BUTTON_R3   HID_KEY_MINUS         // R3     | RS     | RS      | R3       | 12     | RS     |
#define KEY_BUTTON_A1   HID_KEY_9             // A1     | Guide  | Home    | PS       | 13     | ~      |
#define KEY_BUTTON_A2   HID_KEY_F2            // A2     | ~      | Capture | ~        | 14     | ~      |
#define KEY_BUTTON_FN   -1                    // Hotkey Function                                        |

#define USB_PERIPHERAL_ENABLED 1
#define USB_PERIPHERAL_PIN_DPLUS 3
#define USB_PERIPHERAL_PIN_ORDER 1

#define DEFAULT_INPUT_MODE INPUT_MODE_PS5
#define DEFAULT_PS5AUTHENTICATION_TYPE INPUT_MODE_AUTH_TYPE_USB
#define DEFAULT_INPUT_MODE_B4 INPUT_MODE_PS5

#define BOARD_LEDS_PIN 21

#define LED_BRIGHTNESS_MAXIMUM 50
#define LEDS_PER_PIXEL 2
#define LEDS_BASE_ANIMATION_INDEX 1

#define LEDS_BUTTON_R3   0
#define LEDS_BUTTON_A2   0
#define LEDS_DPAD_UP     0
#define LEDS_DPAD_UP     0
#define LEDS_BUTTON_R3   0
#define LEDS_BUTTON_L2   0
#define LEDS_BUTTON_R2   0
#define LEDS_BUTTON_B2   0
#define LEDS_BUTTON_B1   0
#define LEDS_BUTTON_B3   0
#define LEDS_BUTTON_B4   0
#define LEDS_BUTTON_R1   0
#define LEDS_BUTTON_L1   0
#define LEDS_BUTTON_L3   0
#define LEDS_DPAD_RIGHT  0
#define LEDS_DPAD_DOWN   0
#define LEDS_DPAD_LEFT   0
#define LEDS_DPAD_UP     0

//Data format = {first led index, leds on this light, xcoord, ycoord, GPIO pin/case light Index, Type}
//Eg,  your first light would be "first led index" = 0 and "leds on this light" = 2.
//      your second light would be "first led index" = 2 (as 0 and 1 were just taken by the first light)
#define LIGHT_DATA_SIZE 18 //number of sets in the below data
#define LIGHT_DATA \
0, 2, 1, 3, 22, LightType::LightType_ActionButton, \
2, 2, 4, 6, 20, LightType::LightType_ActionButton, \
4, 2, 6, 6, 19, LightType::LightType_ActionButton, \
6, 2, 7, 6, 16, LightType::LightType_ActionButton, \
8, 2, 9, 6, 17, LightType::LightType_ActionButton, \
10, 2, 12, 4, 18, LightType::LightType_ActionButton, \
12, 2, 10, 4, 15, LightType::LightType_ActionButton, \
14, 2, 9, 4, 14, LightType::LightType_ActionButton, \
16, 2, 8, 4, 9, LightType::LightType_ActionButton, \
18, 2, 8, 3, 10, LightType::LightType_ActionButton, \
20, 2, 9, 2, 11, LightType::LightType_ActionButton, \
22, 2, 10, 3, 12, LightType::LightType_ActionButton, \
24, 2, 12, 3, 13, LightType::LightType_ActionButton, \
26, 2, 8, 2, 4, LightType::LightType_ActionButton, \
28, 2, 5, 4, 29, LightType::LightType_ActionButton, \
30, 2, 4, 3, 27, LightType::LightType_ActionButton, \
32, 2, 3, 3, 26, LightType::LightType_ActionButton, \
34, 2, 4, 2, 28, LightType::LightType_ActionButton

#define HAS_I2C_DISPLAY 1
#define I2C0_ENABLED 1
#define I2C0_PIN_SDA 0
#define I2C0_PIN_SCL 1
#define DISPLAY_I2C_BLOCK i2c0
#define DISPLAY_FLIP 0

#define REVERSE_UP_DEFAULT 1
#define REVERSE_DOWN_DEFAULT 1
#define REVERSE_LEFT_DEFAULT 1
#define REVERSE_RIGHT_DEFAULT 1

#define BUTTON_LAYOUT BUTTON_LAYOUT_BOARD_DEFINED_A
#define BUTTON_LAYOUT_RIGHT BUTTON_LAYOUT_BOARD_DEFINED_B
#define SPLASH_MODE SPLASH_MODE_STATIC
#define SPLASH_DURATION 3000

#define FOCUS_MODE_ENABLED 1
#define FOCUS_MODE_BUTTON_MASK GAMEPAD_MASK_S1 | GAMEPAD_MASK_S2 | GAMEPAD_MASK_A1
#define FOCUS_MODE_BUTTON_LOCK_ENABLED 1

#define DEFAULT_DEBOUNCE_DELAY 10

#define DEFAULT_BOARD_LAYOUT_A {\
	{GP_ELEMENT_PIN_BUTTON, {12, 32, 5, 5, 1, 1, 22,   GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_PIN_BUTTON, {25, 29, 5, 5, 1, 1, 26,   GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_PIN_BUTTON, {38, 29, 5, 5, 1, 1, 27,   GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_PIN_BUTTON, {41, 56, 6, 6, 1, 1, 20,   GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_PIN_BUTTON, {43, 16, 5, 5, 1, 1, 28,   GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_PIN_BUTTON, {50, 35, 5, 5, 1, 1, 29,   GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_PIN_BUTTON, {57, 57, 6, 6, 1, 1, 19,   GP_SHAPE_ELLIPSE}},\
	{GP_ELEMENT_PIN_BUTTON, {18, 13, 2, 2, 1, 1, 23,   GP_SHAPE_ELLIPSE}},\
	{GP_ELEMENT_PIN_BUTTON, {24, 13, 2, 2, 1, 1, 24,   GP_SHAPE_ELLIPSE}},\
	{GP_ELEMENT_PIN_BUTTON, {30, 13, 2, 2, 1, 1, 25,   GP_SHAPE_ELLIPSE}}\
}

#define DEFAULT_BOARD_LAYOUT_B {\
    {GP_ELEMENT_PIN_BUTTON, {72, 57, 6, 6, 1, 1, 16,   GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_PIN_BUTTON, {77, 43, 5, 5, 1, 1, 9,    GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_PIN_BUTTON, {78, 30, 5, 5, 1, 1, 10,   GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_PIN_BUTTON, {79, 17, 5, 5, 1, 1, 4,    GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_PIN_BUTTON, {87, 56, 6, 6, 1, 1, 17,   GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_PIN_BUTTON, {89, 37, 5, 5, 1, 1, 14,   GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_PIN_BUTTON, {90, 24, 5, 5, 1, 1, 11,   GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_PIN_BUTTON, {102, 38, 5, 5, 1, 1, 15,  GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_PIN_BUTTON, {103, 25, 5, 5, 1, 1, 12,  GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_PIN_BUTTON, {115, 40, 5, 5, 1, 1, 18,  GP_SHAPE_ELLIPSE}},\
    {GP_ELEMENT_PIN_BUTTON, {116, 27, 5, 5, 1, 1, 13,  GP_SHAPE_ELLIPSE}},\
	{GP_ELEMENT_PIN_BUTTON, {98, 13, 2, 2, 1, 1, 6,    GP_SHAPE_ELLIPSE}},\
	{GP_ELEMENT_PIN_BUTTON, {104, 13, 2, 2, 1, 1, 7,   GP_SHAPE_ELLIPSE}},\
	{GP_ELEMENT_PIN_BUTTON, {110, 13, 2, 2, 1, 1, 8,   GP_SHAPE_ELLIPSE}}\
}

#define DEFAULT_SPLASH \
0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x07, 0xff, 0xf8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x07, 0xff, 0xfc, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x07, 0xff, 0xfc, 0x00, 0x03, 0xff, 0x00, 0x00, 0x00, 0x00, 0x03, 0xfe, 0x00, 0x00, 0x00, \
	0x00, 0x07, 0xff, 0xfc, 0x00, 0x0f, 0xfe, 0x00, 0x00, 0x00, 0x00, 0x07, 0xfc, 0x00, 0x00, 0x00, \
	0x00, 0x07, 0xff, 0xfc, 0x00, 0x1f, 0xfc, 0x00, 0x00, 0x00, 0x00, 0x0f, 0xf8, 0x00, 0x00, 0x00, \
	0x00, 0x07, 0xff, 0xfc, 0x00, 0x3f, 0xf8, 0x00, 0x00, 0x00, 0x00, 0x3f, 0xf0, 0x38, 0x00, 0x00, \
	0x00, 0x07, 0xff, 0xfc, 0x00, 0x3c, 0x00, 0x7f, 0x00, 0x02, 0x1f, 0xfc, 0x00, 0xff, 0x06, 0x3e, \
	0x00, 0x07, 0xff, 0xfc, 0x00, 0x3c, 0x00, 0xff, 0x88, 0x06, 0x3f, 0xfc, 0x01, 0xff, 0x8f, 0x7c, \
	0x00, 0x07, 0xff, 0xfc, 0x00, 0x3e, 0x01, 0xff, 0x98, 0x0e, 0x7f, 0x3c, 0x03, 0xff, 0x9f, 0xf8, \
	0x00, 0x07, 0xff, 0xfc, 0x00, 0x3f, 0xe3, 0x87, 0xb8, 0x1e, 0x0e, 0x3f, 0xe7, 0x83, 0xcf, 0xf0, \
	0x00, 0x07, 0xff, 0xfc, 0x00, 0x1f, 0xf3, 0x83, 0xb8, 0x1e, 0x0e, 0x3f, 0xe7, 0x01, 0xc7, 0xe0, \
	0x00, 0x07, 0xff, 0xfc, 0x00, 0x0f, 0xf3, 0x87, 0xb8, 0x1e, 0x0e, 0x3f, 0xe7, 0x01, 0xc7, 0xf0, \
	0x00, 0x07, 0xff, 0xfc, 0x00, 0x00, 0xfb, 0x9f, 0xb8, 0x1e, 0x0e, 0x3c, 0x07, 0x01, 0xcf, 0xf8, \
	0x00, 0x07, 0xff, 0xfc, 0x00, 0x00, 0xfb, 0x9f, 0x38, 0x1e, 0x0e, 0x3c, 0x07, 0x01, 0xdf, 0xf8, \
	0x00, 0x07, 0xff, 0xfc, 0x00, 0x00, 0xfb, 0x9e, 0x38, 0x1e, 0x0e, 0x3c, 0x07, 0x83, 0xde, 0x78, \
	0x00, 0x0f, 0xff, 0xfc, 0x00, 0x3f, 0xf3, 0x80, 0x3f, 0xde, 0x0e, 0x3c, 0x03, 0xff, 0x9c, 0x38, \
	0x00, 0x0f, 0xff, 0xfe, 0x00, 0x3f, 0xf3, 0x80, 0x3f, 0xde, 0x0e, 0x3c, 0x01, 0xff, 0x18, 0x18, \
	0x00, 0x0f, 0xff, 0xfe, 0x00, 0x3f, 0xe3, 0x80, 0x3f, 0xde, 0x0e, 0x3c, 0x00, 0xfe, 0x10, 0x08, \
	0x00, 0x1f, 0xff, 0xff, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x1f, 0xff, 0xff, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x3f, 0xff, 0xff, 0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x3f, 0xff, 0xff, 0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x7f, 0xff, 0xff, 0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x7f, 0xff, 0xff, 0xc0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x7f, 0xff, 0xff, 0xc0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0xff, 0xff, 0xff, 0xe0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0xff, 0xff, 0xff, 0xe0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x01, 0xff, 0xff, 0xff, 0xf0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x01, 0xff, 0xff, 0xff, 0xf0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x03, 0xff, 0xff, 0xff, 0xf8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x03, 0xff, 0xff, 0xff, 0xf8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x03, 0xff, 0xff, 0xff, 0xf8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x07, 0xff, 0xff, 0xff, 0xfc, 0x00, 0x00, 0x00, 0x1f, 0xc0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x07, 0xff, 0xff, 0xff, 0xfc, 0x00, 0x00, 0x00, 0x3f, 0xf0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x0f, 0xff, 0xff, 0xff, 0xfe, 0x00, 0x00, 0x00, 0x7f, 0xf8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x0f, 0xff, 0xff, 0xff, 0xfe, 0x00, 0x00, 0x00, 0xff, 0xf8, 0x00, 0x01, 0xe0, 0x00, 0x00, 0x00, \
	0x1f, 0xff, 0xfe, 0x00, 0xff, 0x00, 0x00, 0x01, 0xe0, 0xf8, 0x7f, 0x07, 0xf8, 0x00, 0x00, 0x00, \
	0x1f, 0xff, 0xf8, 0x00, 0x0f, 0x00, 0x00, 0x01, 0xe0, 0xf8, 0xff, 0x8f, 0xfc, 0x00, 0x00, 0x00, \
	0x3f, 0xff, 0xff, 0x00, 0x07, 0x80, 0x00, 0x01, 0xe0, 0x79, 0xcf, 0x9f, 0x7e, 0x00, 0x00, 0x00, \
	0x3f, 0xff, 0xff, 0xf8, 0x01, 0x80, 0x00, 0x01, 0xe0, 0xff, 0x83, 0xbc, 0x1e, 0x00, 0x00, 0x00, \
	0x3f, 0xff, 0xff, 0xff, 0xc1, 0x80, 0x00, 0x01, 0xe7, 0xfb, 0x83, 0xb8, 0x0e, 0x00, 0x00, 0x00, \
	0x79, 0xff, 0xff, 0xff, 0xf9, 0xc0, 0x00, 0x01, 0xe7, 0xfb, 0x87, 0xb8, 0x0e, 0x00, 0x00, 0x00, \
	0x78, 0x7f, 0xff, 0xff, 0xff, 0xc0, 0x00, 0x01, 0xe7, 0xf3, 0x8f, 0x38, 0x0e, 0x00, 0x00, 0x00, \
	0xf8, 0x1f, 0xff, 0xff, 0xff, 0xe0, 0x00, 0x01, 0xe7, 0xc3, 0x87, 0x3c, 0x0e, 0x00, 0x00, 0x00, \
	0x7c, 0x03, 0xff, 0xff, 0xff, 0xc0, 0x00, 0x01, 0xe0, 0x03, 0x83, 0x9e, 0x1e, 0x00, 0x00, 0x00, \
	0x7e, 0x00, 0x1f, 0xff, 0xff, 0xc0, 0x00, 0x01, 0xe0, 0x03, 0x81, 0x9f, 0xfc, 0x00, 0x00, 0x00, \
	0x3f, 0x00, 0x00, 0x3f, 0xff, 0x80, 0x00, 0x01, 0xe0, 0x03, 0x80, 0x8f, 0xf8, 0x00, 0x00, 0x00, \
	0x1f, 0xc0, 0x00, 0x7f, 0xff, 0x00, 0x00, 0x01, 0xe0, 0x03, 0x80, 0x07, 0xf0, 0x00, 0x00, 0x00, \
	0x0f, 0xfc, 0x01, 0xff, 0xfe, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x01, 0xff, 0xff, 0xff, 0xf0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x0f, 0xff, 0xfe, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, \
	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00

#endif
